### PR TITLE
Feature: Linting, Code Formatting, Pre-commit hooks

### PR DESCRIPTION
**Pull request by FaLxy**
This PR adds Linting support to the repository using ESLint, code formatting by using prettier and Pre-commit hook by using Husky.

This PR **DOES NOT** enable linting in pre-commit hook because the code is not function at the time of this PR. The relevant code can be uncommented when the time comes.

### CODE REVIEW

**Comment by GetPsyched** [comment]
Rebase on master, the branch is a commit behind. (diff includes that commit's contents which is weird.)

Also, rationale for Husky? AFAIK pre-commit hooks work without deps.

**Comment by FaLxy** [comment]
I think i squashed incorrectly.
and do you mean by "deps"?

**Comment by GetPsyched** [comment]
> and what do you mean by "deps"?

Dev dependency I guess? I'm unsure what Husky does.

**Comment by FaLxy** [comment]
`Husky enhances your commits and more 🐶 woof!
Automatically lint your commit messages, code, and run tests upon committing or pushing.`

https://typicode.github.io/husky/

File: .husky/pre-commit
Comment (FaLxy) [inline .husky/pre-commit:1]: @GetPsyched check this

**Review by GetPsyched — DISMISSED** [review]

**Comment by GetPsyched** [comment]
@FaLxy, can we uncomment those lines since the code exists?

**Pull request merged**